s3tests_boto3/functional/test_s3select.py

```diff
@@ -449,6 +449,44 @@ def test_schema_definition():
 
     assert res_multiple_defintion.find("multiple definition of column {c4} as schema-column and alias") > 0
 
+@attr('s3select')
+def test_when_than_else_expressions():
+
+    csv_obj = create_random_csv_object(1,10)
+
+    csv_obj_name = get_random_string()
+    bucket_name = "test"
+    upload_csv_object(bucket_name,csv_obj_name,csv_obj)
+
+    res_s3select = remove_xml_tags_from_result(  run_s3select(bucket_name,csv_obj_name,'select  case when ((4*3)==(12)) than "case_1_2" else "case_2_1" end from stdin where (3*3==9);')  ).replace("\n","")          
+
+    nose.tools.assert_equal( res_s3select, "case_1_2,")
+
+@attr('s3select')
+def test_coalesce_expressions():
+
+    csv_obj = create_random_csv_object(1,10)
+
+    csv_obj_name = get_random_string()
+    bucket_name = "test"
+    upload_csv_object(bucket_name,csv_obj_name,csv_obj)
+
+    res_s3select = remove_xml_tags_from_result(  run_s3select(bucket_name,csv_obj_name,'select coalesce(nullif(5,5),nullif(1,1.0)) from stdin;')  ).replace("\n","")   
+
+    nose.tools.assert_equal( res_s3select, "null,")
+
+    res_s3select = remove_xml_tags_from_result(  run_s3select(bucket_name,csv_obj_name,'select coalesce(5, 3) from stdin;')  ).replace("\n","")   
+
+    nose.tools.assert_equal( res_s3select, "5,")
+
+    res_s3select = remove_xml_tags_from_result(  run_s3select(bucket_name,csv_obj_name,'select coalesce(nullif(5,5),nullif(1,1.0),2) from stdin;')  ).replace("\n","")   
+
+    nose.tools.assert_equal( res_s3select, "2,")
+
+    res_s3select = remove_xml_tags_from_result(  run_s3select(bucket_name,csv_obj_name,'select count(*) from stdin where  coalesce(nullif(7.0,7),nullif(4,4.0),6) == 6;')  ).replace("\n","")   
+
+    nose.tools.assert_equal( res_s3select, "1,")
```

Comment:
ok, 
beside the constant values, we had discussed.
the **coalesce** should test for **order of processing**, meaning,
in case it got more than one non-null values , is it return the correct one?

Reply:
Yes, I have tried some unit tests. It always returns the first non-null value